runlog.c

```diff
@@ -312,6 +312,7 @@ run_add_record(
         int            eoln_type,
         int            variant,
         int            is_hidden,
+        int            upsolving,
```

Comment:
I don't see any nice way to check if user is upsolving inside this function. The only way I see to make it without additional parameter is to check it before calling run_add_record, but there are 7 usages of this function in code, so it won't be nice, I think.